llvm/tools/llvm-split/llvm-split.cpp

```diff
@@ -70,6 +73,65 @@ static cl::opt<std::string>
     MCPU("mcpu", cl::desc("Target CPU, ignored if -mtriple is not used"),
          cl::value_desc("cpu"), cl::cat(SplitCategory));
 
+cl::opt<IRSplitMode> SYCLSplitMode(
+    "sycl-split", cl::desc("module split mode"), cl::Optional,
+    cl::init(SPLIT_NONE),
+    cl::values(
+        clEnumValN(SPLIT_PER_TU, "source",
+                   "1 output module per source (translation unit)"),
+        clEnumValN(SPLIT_PER_KERNEL, "kernel", "1 output module per kernel"),
+        clEnumValN(SPLIT_AUTO, "auto", "Choose split mode automatically")),
+    cl::cat(SplitCategory));
+
+cl::opt<bool> OutputAssembly{"S", cl::desc("Write output as LLVM assembly"),
```

Comment:
Is this support necessary?

Reply:
It simplifies the testing and debugging. Otherwise, you need to use `llvm-dis` tool to convert `bc` file to `ll` format. I see the problem that this option works only for SYCL while it will be asked to be supported for other use cases as well.